### PR TITLE
Add SSO UI

### LINKED ISSUE #619: Make the onboarding (welcome) screen configurable for SSO
## Description
Add a flag in state to tell the onboarding view how to configure itself. Then make the UI respect that flag. Details https://github.com/toggl/mobileapp/issues/7420 and https://github.com/toggl/mobileapp/issues/7421. 

The transition between the two states has to happen with a smooth animation (check the Xamarin app).

## ~~Dependencies~~ Definition of done
Flag is added to state
View is configured according to the flag (this includes adding the missing pieces, like the cancel button)
Transition between views (after hitting cancel), happens smoothly

### LINKED ISSUE #621: Add 'Login with SSO' button and 'Login with SSO' view
## Description
Add login with SSO button to the main onboarding screen. Add the login with SSO view. Details https://github.com/toggl/mobileapp/issues/7412 https://github.com/toggl/mobileapp/issues/7411
## Definition of done
Button is added to the main screen
SSO login view is added

### LINKED ISSUE #622: Add LoginWithSsoButtonTapped action
## Description
Add the action and reduce it so that navigation to ‘Login with SSO view’ happens
## Definition of done
Action is added
It’s wired to the UI
Navigation happens
Tests are added
- [ ] The documentation is updated to reflect the changes done in this PR

### LINKED ISSUE #624: Add CancelButtonTapped action
## Description
Add CancelButtonTapped action https://www.notion.so/toggl/Welcome-63c8191fa8144fd5af36a2f266cad16a?p=330d4694b9794e80b837e714ddeb73b6 
## Dependencies
#619 

## Definition of done
Action is added
It's wired to the UI
It’s reduced so that the onboarding screen is reconfigured to show the regular state
Tests are added

- [ ] The documentation is updated to reflect the changes done in this PR

### LINKED ISSUE #625: SSO Add email entered action
## Description
Add email entered action for the sso module. Wire it up to the ui. Info https://www.notion.so/toggl/SSO-d8929cc996084f3db6fdd08637080f1c

## Dependencies
#621 
## Definition of done
Action is added
Tests are added
UI is wired up

- [ ] The documentation is updated to reflect the changes done in this PR

### LINKED ISSUE #627: Add SSO continue button tapped action
## Description
Add continue button tapped action for the sso module. This should emit GetSsoUrlEffect and forward the email
## Dependencies
#649 
## Definition of done
Action is added
Tests are added
UI is wired up

- [ ] The documentation is updated to reflect the changes done in this PR

### LINKED ISSUE #635: Add LinkSsoButtonTapped action
## Description
Add the action. Reduce it so that user is taken to the custom welcome screen prepared in #619. The email and confirmation code params have to be preserved somewhere so they can be used in effect later
## Dependencies
#619 
#618 
## Definition of done
Action is added, reduced, tested
UI is wired up

- [ ] The documentation is updated to reflect the changes done in this PR

### LINKED ISSUE #636: Add CloseLinkSsoTapped action
## Description
Add the action, reduce it so that user is taken back to the main welcome screen.

## Dependencies
#618  

## Definition of done
Action is added, reduced, tested
UI is wired up

- [ ] The documentation is updated to reflect the changes done in this PR

### PR DESCRIPTION
**Pull request by dshr**
### Summary
This PR adds all SSO UI screens and some required interactions.

### Technical considerations
I've made the SSOCoordinator a subclass of NavigationCoordinator, which means it's root view controller is a UINavigationController. This means that it needs it's own `UIAdaptivePresentationControllerDelegate` that will handle the swipe-to-dismiss gesture. I ended up using the `LoginWithSSOViewController` as that delegate, since it's the root view controller of that UINavigationController. LMK if there's a better way.

### Relationships
Closes #636
Closes #635
Closes #627
Closes #625
Closes #624
Closes #622
Closes #621
Closes #619

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

### Pointers
Commit by commit y'all

### Review template
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones
- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Go for it!

### CODE REVIEW

**Comment by dshr** [comment]
Forgot to add localisations, give me a sec

File: Modules/Assets/Assets/Base.lproj/Localizable.strings [hunk 1]
```diff
@@ -117,3 +117,12 @@
...
 "linkAccountCTA" = "Log in to enable SSO";
+"SSOTitle" = "Single sign on";
+"workEmail" = "Work Email";
+"workEmailPlaceholder" = "Enter your work email";
+"linkAccountsDescription" = "Please log in with your existing credentials";
+"thirdPartyLoginPrompt" = "Or continue using";
+"thirdPartyLoginCTAFull" = "Apple, Google or SSO";
+"thirdPartyLoginCTASSO" = "Apple or Google";
+"continue" = "Continue";
+"continueWithEmail" = "Continue with email";
```
Comment (sofokli1) [inline Modules/Assets/Assets/Base.lproj/Localizable.strings:128]: ref #715

**Review by sofokli1 — DISMISSED** [review]
didn't review all of it since it it getting a bit late (will continue tomorrow with the rest) but until than here are a couple of issues I found.

1. In the initial view the border of the email field does not change color. (would you consider this lack of functionality or pixel perfect rabbit hole that we said to postpone?)
2. - Open the first sso view
- type an email and continue
- tap `x` button in the new view
- open the sso again, type an email, tap continue
- not going anywhere now (unless I restart the app)

**Review by sofokli1 — APPROVED** [review]
looks good now

**Pull request merged**